components/Filter/FilterBy/Trait.tsx

```diff
@@ -11,54 +14,96 @@ import {
   Stack,
   Text,
   VStack,
+  useToast,
 } from '@chakra-ui/react'
+import SearchInput from 'components/SearchInput'
 import useTranslation from 'next-translate/useTranslation'
-import { FC, useCallback, useMemo } from 'react'
+import { FC, useCallback } from 'react'
 import { UseFormReturn } from 'react-hook-form'
-import { useFetchCollectionTraitsQuery } from '../../../graphql'
+import {
+  CollectionTraitFilter,
+  StringFilter,
+  useFetchCollectionTraitsQuery,
+} from '../../../graphql'
 import { Filter } from '../../../hooks/useAssetFilterFromQuery'
+import { formatError } from '../../../utils'
 
 type Props = {
-  collection:
-    | {
-        chainId: number
-        address: string
-      }
-    | undefined
-  filter: Filter
+  collection: {
+    chainId: number
+    address: string
+  }
   formValues: UseFormReturn<Filter, any, undefined>
   onFilterChange: (filter: Filter) => void
 }
 
+const PAGINATION_LIMIT = 8
+
 const FilterByTrait: FC<Props> = ({
   collection,
-  filter,
-  formValues: { watch },
+  formValues: { setValue, watch },
   onFilterChange,
 }) => {
   const { t } = useTranslation('components')
+  const toast = useToast()
 
   const filterResult = watch()
 
-  const { data } = useFetchCollectionTraitsQuery({
+  const { data, fetchMore, networkStatus } = useFetchCollectionTraitsQuery({
     variables: {
-      address: (collection && collection.address) || '',
-      chainId: (collection && collection.chainId) || 0,
+      address: collection.address,
+      chainId: collection.chainId,
+      cursor: null,
+      limit: PAGINATION_LIMIT,
+      filter: {
+        type: {
+          includesInsensitive: filterResult.propertySearch || '',
+        } as StringFilter,
+      } as CollectionTraitFilter,
```

Comment:
Could you try to remove the fallback value on `filterResult.propertySearch` and not pass any filter, something like this:
```suggestion
      filter: filterResult.propertySearch ? {
        type: {
          includesInsensitive: filterResult.propertySearch,
        } as StringFilter,
      } : {} as CollectionTraitFilter,
```
This will better in term of SQL query executed by the backend